contracts/managers/SocialTradingManagerV2.sol

```diff
@@ -0,0 +1,104 @@
+import { IRebalancingSetTokenV2 } from "set-protocol-contracts/contracts/core/interfaces/IRebalancingSetTokenV2.sol";
+import { IRebalancingSetTokenV3 } from "set-protocol-contracts/contracts/core/interfaces/IRebalancingSetTokenV3.sol";
+import { LimitOneUpgrade } from "set-protocol-contracts/contracts/lib/LimitOneUpgrade.sol";
+
+import { SocialTradingManager } from "./SocialTradingManager.sol";
+
+
+/**
+ * @title SocialTradingManagerV2
+ * @author Set Protocol
+ *
+ * Singleton manager contract through which all social trading v2 sets are managed. Inherits from SocialTradingManager
+ * and adds functionality to adjust performance based fees.
+ */
+contract SocialTradingManagerV2 is
+    SocialTradingManager,
+    LimitOneUpgrade
+{
+    /*
+     * SocialTradingManager constructor.
+     *
+     * @param  _core                            The address of the Core contract
+     * @param  _factory                         Factory to use for RebalancingSetToken creation
+     * @param  _whiteListedAllocators           List of allocator addresses to WhiteList
+     * @param  _maxEntryFee                     Max entry fee when updating fees in a scaled decimal value
+     *                                          (e.g. 1% = 1e16, 1bp = 1e14)
+     * @param  _feeUpdateTimelock               Amount of time trader must wait between starting fee update
+     *                                          and finalizing fee update
+     */
+    constructor(
+        ICore _core,
+        address _factory,
+        address[] memory _whiteListedAllocators,
+        uint256 _maxEntryFee,
+        uint256 _feeUpdateTimelock
+    )
+        public
+        SocialTradingManager(
+            _core,
+            _factory,
+            _whiteListedAllocators,
+            _maxEntryFee,
+            _feeUpdateTimelock
+        )
+    {}
+
+    /* ============ External ============ */
+
+    /**
+     * Allows traders to update fees on their Set. Only one fee update allowed at a time and timelocked.
+     *
+     * @param _tradingPool       The address of the trading pool being updated
+     * @param _newFeeCallData    Bytestring representing feeData to pass to fee calculator
+     */
+    function adjustFee(
+        address _tradingPool,
+        bytes calldata _newFeeCallData
+    )
+        external
+        onlyTrader(IRebalancingSetTokenV2(_tradingPool))
+        limitOneUpgrade(_tradingPool)
+        timeLockUpgrade
+    {
+        IRebalancingSetTokenV3(_tradingPool).adjustFee(_newFeeCallData);
+    }
+
+    /**
+     * External function to remove upgrade. Modifiers should be added to restrict usage.
+     *
+     * @param _tradingPool      The address of the trading pool being updated
+     * @param _upgradeHash      Keccack256 hash that uniquely identifies function called and arguments
+     */
+    function removeRegisteredUpgrade(
+        address _tradingPool,
+        bytes32 _upgradeHash
```

Comment:
If the only check is that the trading pool is the trader's, can't an arbitrary trader remove someone else's upgradehash?

Reply:
It seems there needs to be a require that the `upgradeHash` is mapped to the right `tradingPool`

Reply:
What I would recommend is in the `LimitOneUpgrade` file, that the `upgradeInProgress` mapping be `address` => upgradeHash (`bytes32) instead of boolean to express specifically what upgrade is in progress.

Reply:
Also the comment on Line 52 of `LimitOneUpgrade` needs to be updated.